core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -424,6 +438,10 @@ private void installConfig(PluginInfo info, Path tmpConfigDir, Path destConfigDi
         // create the plugin's config dir "if necessary"
         Files.createDirectories(destConfigDir);
 
+        final PosixFileAttributes destConfigDirAttributes =
+                Files.getFileAttributeView(destConfigDir.getParent(), PosixFileAttributeView.class).readAttributes();
+        setOwnerGroup(destConfigDir, destConfigDirAttributes);
```

Comment:
Note that this only applies to the plugin config folder. The packaging creates the config folders with owner `root` and group `elasticsearch`; this code ensures that a plugin config folder is created with the same owner and group as the parent config folder (otherwise it will come out `root:root`).